tests/e2e/vsan_stretched_cluster.go

```diff
@@ -363,4 +376,482 @@ var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", f
+		for _, sspod := range ssPodsBeforeScaleDown.Items {
+			_, err := client.CoreV1().Pods(namespace).Get(ctx, sspod.Name, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, volumespec := range sspod.Spec.Volumes {
+				if volumespec.PersistentVolumeClaim != nil {
+					pv := getPvFromClaim(client, statefulset.Namespace, volumespec.PersistentVolumeClaim.ClaimName)
+					// Verify the attached volume match the one in CNS cache
+					err := verifyVolumeMetadataInCNS(&e2eVSphere, pv.Spec.CSI.VolumeHandle,
+						volumespec.PersistentVolumeClaim.ClaimName, pv.ObjectMeta.Name, sspod.Name)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}
+
+		statefulset1 := GetStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		statefulset1.Spec.VolumeClaimTemplates[len(statefulset1.Spec.VolumeClaimTemplates)-1].
+			Annotations["volume.beta.kubernetes.io/storage-class"] = storageClassName
+		statefulset1.Name = "web-nginx"
+		statefulset1.Spec.Template.Labels["app"] = statefulset1.Name
+		statefulset1.Spec.Selector.MatchLabels["app"] = statefulset1.Name
+		*(statefulset1.Spec.Replicas) = 5
+		CreateStatefulSet(namespace, statefulset1, client)
+		replicas1 := *(statefulset1.Spec.Replicas)
+		// Waiting for pods status to be Ready
+		fss.WaitForStatusReadyReplicas(client, statefulset1, replicas1)
+		gomega.Expect(fss.CheckMount(client, statefulset1, mountPath)).NotTo(gomega.HaveOccurred())
+		ss1PodsBeforeScaleDown := fss.GetPodList(client, statefulset1)
+		gomega.Expect(ss1PodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(),
+			fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset1.Name))
+		gomega.Expect(len(ss1PodsBeforeScaleDown.Items) == int(replicas1)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset: %v should match with number of replicas: %v", statefulset1.Name, replicas1)
+
+		// Get the list of Volumes attached to Pods of sts2 before scale down
+		for _, ss1pod := range ss1PodsBeforeScaleDown.Items {
+			_, err := client.CoreV1().Pods(namespace).Get(ctx, ss1pod.Name, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, volumespec1 := range ss1pod.Spec.Volumes {
+				if volumespec1.PersistentVolumeClaim != nil {
+					pv := getPvFromClaim(client, statefulset1.Namespace, volumespec1.PersistentVolumeClaim.ClaimName)
+					// Verify the attached volume match the one in CNS cache
+					err := verifyVolumeMetadataInCNS(&e2eVSphere, pv.Spec.CSI.VolumeHandle,
+						volumespec1.PersistentVolumeClaim.ClaimName, pv.ObjectMeta.Name, ss1pod.Name)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}
+
+		ginkgo.By("Bring down the primary site")
+		siteFailover(true)
+
+		ginkgo.By("Wait for k8s cluster to be healthy")
+		wait4AllK8sNodesToBeUp(ctx, client, nodeList)
+		err = waitForAllNodes2BeReady(ctx, client, pollTimeout*4)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		replicas += 2
```

Comment:
statefulset scaling should be started and when that is happening site failure should be triggered
you can use more number of statefulsets for the test